nomad/resource_acl_policy.go

```diff
@@ -43,10 +44,78 @@ func resourceACLPolicy() *schema.Resource {
 				Required:    true,
 				Type:        schema.TypeString,
 			},
+
+			"workload_identity": {
+				Description: "Workload identity that this should be applied to.",
+				Optional:    true,
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"namespace": {
+							Description: "Namespace",
+							Type:        schema.TypeString,
+							Optional:    true,
+							Default:     "default",
+						},
+						"job_id": {
+							Description: "Job",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"group": {
+							Description: "Group",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						"task": {
+							Description: "Task",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+					},
+				},
+			},
 		},
 	}
 }
 
+func parseWorkloadIdentity(workloadIdentity interface{}) (*api.JobACL, error) {
+	jobPolicyList, ok := workloadIdentity.([]interface{})
+	if !ok || len(jobPolicyList) != 1 {
+		return nil, errors.New("failed to unpack workload_identity configuration block")
```

Comment:
```suggestion
		return nil, errors.New("only one workload_identity block allowed")
```

It's better to give specific guidance and context on what the error was and keep them different from the error message below so we know which code path was hit when debugging problems 😄